WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsViewModel.kt

```diff
@@ -129,11 +129,17 @@ class SiteCreationIntentsViewModel @Inject constructor(
     }
 
     fun onSearchTextChanged(query: String) {
-        val searchResults = searchResultsProvider.search(fullItemsList.items, query)
+        val searchResults = searchResultsProvider.search(fullItemsList.items, query).toMutableList().apply {
+            if (query.isNotEmpty() && !(this.size == 1 && this[0].verticalText.equals(query, true))) {
+                val customVertical = IntentListItemUiState("", query, "+").apply {
```

Comment:
np: It might be possible to extract this into a static `IntentListItemUiState` object; but I haven't done the mental exercise to see if that's easy.. thus not really pushing for it 😇

Reply:
Seems like a good idea. Same as: https://github.com/wordpress-mobile/WordPress-Android/pull/16232/files#r841286602, we can make it non-blocking, and address it in a followup PR.

Reply:
> It might be possible to extract this into a static IntentListItemUiState object

I thought of using a static object but I felt that this might complicate things since the `val verticalText: String` need to change. To overcome this I could think of two options:
* make the val a variable which would beat the purpose of the object
* use the static object as a kind of flag and get the latest query from another source. I thought that this would make the code too complicated

Wdyt?

Reply:
Added another suggestion to this with https://github.com/wordpress-mobile/WordPress-Android/pull/16242/commits/223202a9ac2095a3dd118342a54f23249e04b29a